File: erpnext/gp_erp/controllers/manufacturing/routing.js
frappe.ui.form.on('Routing', {
	calculate_operating_cost: function(frm, child) {
		var operating_cost = 0;
		if (child.calculation_type == 'Per Qty') {
			operating_cost = flt(flt(child.operation_rate) * 1, precision("operating_cost", child));
		} else {
			operating_cost = flt(flt(child.operation_rate) * flt(child.time_in_mins) / 60, precision("operating_cost", child));
		}
		frappe.model.set_value(child.doctype, child.name, "operating_cost", operating_cost);
	}
});

frappe.ui.form.on('BOM Operation', {
	workstation: function(frm, cdt, cdn) {
		const d = locals[cdt][cdn];
		if (d.workstation) {
			frappe.call({
				"method": "frappe.client.get",
				args: {
					doctype: "Workstation",
					name: d.workstation
				},
				callback: function (data) {
					var opr_rate = 0;
					if (data.message.calculation_type == 'Per Qty') {
						opr_rate = data.message.per_qty_rate;
					} else {
						opr_rate = data.message.hour_rate;
					}
					frappe.model.set_value(d.doctype, d.name, "operation_rate", flt(opr_rate, 2));
					frm.events.calculate_operating_cost(frm, d);
				}
			});
		} else {
			frappe.model.set_value(d.doctype, d.name, "operation_rate", 0);
			frm.events.calculate_operating_cost(frm, d);
		}
	},
});

File: erpnext/gp_erp/controllers/manufacturing/routing.py
import frappe
from frappe import _
from frappe.utils import flt

from erpnext.manufacturing.doctype.routing.routing import Routing


class RoutingGP(Routing):
    def calculate_operating_cost(self):
        for operation in self.operations:
            if not operation.operation_rate:
                if operation.calculation_type == "Per Qty":
                    operation.operation_rate = frappe.db.get_value("Workstation", operation.workstation, "per_qty_rate")
                else:
                    operation.operation_rate = frappe.db.get_value("Workstation", operation.workstation, "hour_rate")
            if operation.calculation_type == "Per Qty":
                operation.operating_cost = flt(operation.operation_rate)
            else:
                operation.operating_cost = flt(
                    flt(operation.operation_rate) * flt(operation.time_in_mins) / 60,
                    operation.precision("operating_cost"),
                )
